packages/commerce-sdk-react/src/hooks/ShopperOrders/mutation.ts

```diff
@@ -81,19 +80,43 @@ export const SHOPPER_ORDERS_NOT_IMPLEMENTED = [
     'UpdatePaymentInstrumentForOrder'
 ]
 
+export type ShopperOrdersMutationType = typeof ShopperOrdersMutations[keyof typeof ShopperOrdersMutations]
+
+type UseShopperOrdersMutationHeaders = NonNullable<Argument<Client['createOrder']>>['headers']
+type UseShopperOrdersMutationArg = {
+    headers?: UseShopperOrdersMutationHeaders
+    rawResponse?: boolean
+    action: ShopperOrdersMutationType
+}
+
 /**
  * A hook for performing mutations with the Shopper Orders API.
  */
-export function useShopperOrdersMutation<Action extends ShopperOrdersMutationType>(action: Action) {
+function useShopperOrdersMutation<Action extends ShopperOrdersMutationType>(
+    arg: UseShopperOrdersMutationArg & {rawResponse?: false}
+): UseMutationResult<DataType<Client[Action]>, Error>
+function useShopperOrdersMutation<Action extends ShopperOrdersMutationType>(
+    arg: UseShopperOrdersMutationArg & {rawResponse: true}
+): UseMutationResult<Response, Error>
+function useShopperOrdersMutation<Action extends ShopperOrdersMutationType>(
+    arg: UseShopperOrdersMutationArg
+): UseMutationResult<DataType<Client[Action]>, Error, Argument<Client[Action]>> {
+    const {headers, rawResponse, action} = arg
+
     if (SHOPPER_ORDERS_NOT_IMPLEMENTED.includes(action)) {
         NotImplemented()
     }
     type Params = Argument<Client[Action]>
     type Data = DataType<Client[Action]>
     const queryClient = useQueryClient()
+
     return useMutation<Data, Error, Params>(
         (params, apiClients) => {
             const method = apiClients['shopperOrders'][action] as MutationFunction<Data, Params>
+            //Set headers to user-defined header values
+            if (params) {
+                params.headers = headers
```

Comment:
I don't think we need to override the headers at this point 🤔 We already have them at line 104, no?

Reply:
I don't think `headers` at line 104 is connected to `params` inside `useMutation` so overriding is necessary.

Reply:
@vmarta is right, the user already defines the `headers` inside the `arg` object when using the mutation, `params` is the `arg` object.

One way to validate this:
1. Add a `header` in the `useShopperOrdersMutation` call on the test-commerce-sdk-react project:
https://github.com/SalesforceCommerceCloud/pwa-kit/blob/23fd34817ea1cb5c76ab02fa4eeaf91b505a998a/packages/test-commerce-sdk-react/app/pages/use-shopper-orders.tsx#L57

```
            hook: useShopperOrdersMutation( {headers: {test-header: 'value'}, rawResponse: true, action}),
```

2. Check the content of `params`.
https://github.com/SalesforceCommerceCloud/pwa-kit/blob/dbdef0a8afd28065a7e66b1d8fe60d8c4a2b6c25/packages/commerce-sdk-react/src/hooks/ShopperOrders/mutation.ts#L114

Reply:
Ignore my comment, as you say the override is needed since `arg` and `params` are two different objects.


`arg` is the object defined by the user in the mutation declaration:
Example:
```
const mutation =  useShopperOrdersMutation( {headers: {test-header: 'value'}, rawResponse: true, action} )
```
```
{
   headers: {test-header: 'value'}, 
   rawResponse: true, 
   action: 'createOrder'
} 
```

Whereas `params` is the object defined by the user when executing the mutation.
Example:
```
mutation.mutate({parameters, body})
```
```
{
   parameters: {}
   body: {basketId: '8140cd55ffa151713f0d88df1d'}
}
```

Reply:
The current signature adds both the `headers` and `rawResponse` into the mutation declaration inside the `arg` object. i.e.
```
const mutation =  useShopperOrdersMutation( {headers, rawResponse, action} )
```

An alternative approach if want to avoid the override could be:

- Add `rawResponse` to the mutation declaration signature.
```
const mutation =  useShopperOrdersMutation( {action, rawResponse} )
```

- Add `headers` to the mutation execution signature.
```
mutation.mutate({parameters, body, headers})
```